test/test-pages/gcba/platform-client.ts

```diff
@@ -0,0 +1,170 @@
+
+    const host = window.location.host;
+    let redirectUri = 'https://';
+
+    if (host === 'localhost:8443') {
+      redirectUri += host;
+    } else {
+      redirectUri += environment.uri;
+    }
+
+    redirectUri += window.location.pathname;
+    console.debug('Signing out with redirect of: ' + redirectUri);
+
+    client.logout(redirectUri);
+  }
+
+  function authenticate () {
+    const environment = getInputValue('environment');
+    const clientId = (window as any).environments[environment].clientId;
+    initClientEnvironment(environment);
+
+    client.loginImplicitGrant(clientId, window.location.href)
+      .then(() => {
+        const authInfo = JSON.parse(window.localStorage.getItem(`${persitentName}_auth_data`));
+        (client as any).authentications['PureCloud Auth'].accessToken = authInfo.accessToken;
+
+        (window as any).conversationsAPI = new (window as any).platformClientConversationsApi();
+        setAuthTextVisible(true);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  function authenticateFromToken (token) {
+    const environment = getInputValue('environment');
+    initClientEnvironment(environment);
+
+    window.localStorage.setItem(`${persitentName}_auth_data`, JSON.stringify({ accessToken: token }));
+    client.authentications['PureCloud OAuth'].accessToken = token;
+    (window as any).conversationsAPI = new (window as any).platformClient.ConversationsApi();
+
+    setAuthTextVisible(true);
+  }
+
+  function authenticateFromUrlToken () {
+    const urlParams = getCurrentUrlParams();
+    if (urlParams) {
+      authenticateFromToken(urlParams.access_token);
+      manualAuthInput.value = urlParams.access_token;
+    }
+  }
+
+  function updateHashParams (newParams: {[param: string]: string}, mergeParams = true) {
+    let params = newParams;
+
+    if (mergeParams) {
+      params = { ...getCurrentUrlParams(), ...newParams };
+    }
+
+    const hashStr = Object.entries(params)
+      .map(([key, value]) => `${key}=${value}`)
+      .join('&');
+    window.location.hash = `#${hashStr}`;
+    window.location.reload();
+  }
+
+  function addTokenToUrl (submitEvent) {
+    submitEvent.preventDefault();
+
+    const token = manualAuthInput.value;
+
+    if (!token) {
+      console.error('No token found');
+      return;
+    }
+
+    const env = getInputValue('environment');
+    updateHashParams({ env, access_token: token });
+  }
+
+  function addJwtToUrl (submitEvent) {
+    submitEvent.preventDefault();
+
+    const jwt = jwtInput.value;
+
+    if (!jwt) {
+      console.error('No jwt found');
+      return;
+    }
+
+    updateHashParams({ jwt });
+  }
+
+  async function fetchNewJwt () {
+    const accessToken = manualAuthInput.value;
+
+    if (!accessToken) {
+      throw new Error('Must have an access token to');
+    }
+
+    const url = `https://api.${(window as any).environments[getInputValue('environment')].uri}/api/v2/screenrecording/token`;
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9580329)
```suggestion
    const url = `https://api.${(window as unknown).environments[getInputValue('environment')].uri}/api/v2/screenrecording/token`;
```